runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1025,7 +1042,16 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 		} else if (dataSizeInBytes < _omrVM->_arrayletLeafSize) {
 			isValidDataAddress = (dataAddr == (void *)((uintptr_t)arrayPtr + contiguousIndexableHeaderSize()));
 		} else {
-			isValidDataAddress = (dataAddr == NULL);
+			if (isVirtualLargeObjectHeapEnabled()
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+			 || isDoubleMappingEnabled()
+#endif /* defined(J9VM_GC_ENABLE_DOUBLE_MAP) */
+			 ) {
+				isValidDataAddress = isValidDataAddrForDoubleMappedObject;
+			}
+			else {
+				isValidDataAddress = (dataAddr == NULL);
```

Comment:
is this really reachable?

Reply:
this part seems unnecessary if we really use this method in offheap enabled case